src/onboarding.py

```diff
@@ -19,16 +20,22 @@ def __init__(self):
         self.add_item(self.email)
 
     async def on_submit(self, interaction):
+        # Defer (new_user takes long)
+        await interaction.response.defer(ephemeral=True)
+
         name = self.name.value
         email = self.email.value
+        user_id = interaction.user.id
+        username = interaction.user.name
 
+        await new_user(user_id, username, email, name)
```

Comment:
My suggestion is to do it like this:
```suggestion
        user = await ensure_user(interaction.user)
        await save_user_personal_details(user, email, name)
```
This way we will not have to change anything elsewhere (including the new_user function)

Reply:
I create show you what I mean in a separate branch if you want

Reply:
yes 👍

Reply:
I have added this function and approached it a little differently. I am calling `ensure_user` within the `save_user_personal_details` to make it more simple.